backend/my_executors.py

```diff
@@ -1,41 +1,53 @@
-#%%
+# %%
 import torch
 from transformers import BertModel, BertTokenizer
 import re
+from typing import Sequence, List
 
 from jina import Executor, requests
 from jina import Document, DocumentArray
 
+
 class ProtBertExecutor(Executor):
+    """ProtBERT executor: https://huggingface.co/Rostlab/prot_bert"""
+
     def __init__(self, **kwargs):
         super().__init__()
 
-        tokenizer = BertTokenizer.from_pretrained("Rostlab/prot_bert", do_lower_case=False )
+        tokenizer = BertTokenizer.from_pretrained(
+            "Rostlab/prot_bert", do_lower_case=False
+        )
         model = BertModel.from_pretrained("Rostlab/prot_bert")
 
         self.tokenizer = tokenizer
         self.model = model
 
-
     @requests
-    def encode(self, docs, **kwargs):
-        sequences = docs.get_attributes('text')
-        sequences = [re.sub(r"[UZOB]", "X", seq) for seq in sequences]
+    def encode(self, docs: DocumentArray, **kwargs) -> DocumentArray:
+        sequences = self.preprocessing(docs.get_attributes("text"))
+        encoded_inputs = self.tokenizer(sequences, return_tensors="pt")
 
-        encoded_inputs = self.tokenizer(sequences, return_tensors='pt')
         with torch.no_grad():
             outputs = self.model(**encoded_inputs)
             embeds = outputs.last_hidden_state[:, 0, :].detach().numpy()
             for doc, embed in zip(docs, embeds):
                 doc.embedding = embed
 
         return docs
-#%%
+
+    def preprocessing(self, sequences: Sequence[str]) -> List[str]:
```

Comment:
This notation is new to me!

Reply:
You mean the type annotations? It's a relatively new features of Python, and I have noticed that in the Jina source code they use annotated code. So I aligned to the style :)

Reply:
Particularly the arrow notation (-> List[str]) after the parameters. [So it specifies the return type. ](https://medium.com/@thomas_k_r/whats-this-weird-arrow-notation-in-python-53d9e293113) Cool!